### PR TITLE
fix: CGO Memory leak issue in GO Feature server

### PR DESCRIPTION
**Pull request by EXPEbdodla**
# What this PR does / why we need it:
Releases the memory used by [CGOArrowAllocator](https://github.com/apache/arrow/blob/go/v8.0.0/go/arrow/memory/cgo_allocator.go). Not releasing the memory causing the memory leaks. Pods restart after all the memory allocated by container is filled up.

# Which issue(s) this PR fixes:
NA

# Fixes

- Memory leak in GO feature server which is causing the pod restarts after allocated memory filled up
- Sets the response content type to application/json (Before its application/text type)
- Added health check endpoint for GO HTTP feature server

### CODE REVIEW

**Comment by EXPEbdodla** [comment]
pr-integration-tests failure is not related to this code change.

**Comment by tokoko** [comment]
You can either make some dummy commit or maybe one of the maintainers can force a rerun? @franciscojavierarceo @HaoXuAI

**Comment by tokoko** [comment]
@EXPEbdodla All commits need a sign-off. Follow instructions [here](https://github.com/feast-dev/feast/pull/4291/checks?check_run_id=26483385040), `git rebase HEAD~3 --signoff` and then `git push --force-with-lease origin fix_memory_leak`.

**Review by tokoko — APPROVED** [review]
lgtm

**Review by franciscojavierarceo — APPROVED** [review]

**Comment by franciscojavierarceo** [comment]
Hi @EXPEbdodla I wanted to give you a shout out in our newsletter but didn't see your name in your actual GitHub Profile, feel free to let me know or add it to the doc if you wouldn't mind here: https://docs.google.com/document/d/1sKD1Lt3h4G9vzitbgoOwNX9uOrLCjdzw4bJnUIO4ixU/edit

**Comment by EXPEbdodla** [comment]
Thanks @franciscojavierarceo . Added my name to the Google Doc

**Pull request merged**
